spec/formatting.md

```diff
@@ -222,6 +222,61 @@ the following steps are taken:
 
    The form that resolved _operand_ and _option_ values take is implementation-defined.
 
+   A _local-declaration_ binds the resolved value of an _expression_
+   to a _variable_.
+   Thus, the output of one _function_ is potentially the _operand_
+   of another _function_. In other words, formatting functions
+   compose with each other.
+   For example, in
+   ```
+   .input {$n :number minIntegerDigits=3}
+   .local {$n1 :number maxFractionDigits=3}
+   ```
+   the second call to `:number` composes with the first call.
+
+   In addition, selector functions compose with formatting functions
+   in the sense that a selector function's _operand_
+   may be the output of any formatting function.
+
+   Implementations SHOULD provide a means for formatting functions
+   to compose with each other
+   and for formatting functions to compose with selector functions.
+   Implementations that provide a means for defining custom functions
+   SHOULD provide a means for those functions to return values
+   that contain enough information
+   (e.g. the resolved _operand_ and _option_ values
+   that the function was called with)
+   to be used as inputs to subsequent function calls.
+   For example, an implementation in a typed programming language
+   MAY define an interface that custom functions implement.
+   Such an interface SHOULD define an implementation-specific
+   argument type `T` and return type `U` for custom formatting functions
+   such that `U` can be coerced to `T` without loss of information.
+   The type `U`
+   (or a type that `U` can be coerced to without loss of information)
+   SHOULD also be the input type of custom selector functions.
+
+> [!NOTE]
+> In the Tech Preview, the spec leaves the behavior of the previous
+> example implementation-dependent. Supposing that
+> the external input variable `n` is bound to the string `"1"`,
+> and that the implementation formats to a string,
+> the formatted result of the following message:
+>
+> ```
+> .input {$n :number minIntegerDigits=3}
+> .local {$n1 :number maxFractionDigits=3}
+> {{$n1}}
+> ```
+>
+> is implementation-dependent.
+> Depending on whether the options are preserved across
+> the two calls to `:number`, a conformant implementation
+> could produce either "001.000" or "1.000"
+> Feedback from users and implementers is desired
+> about whether to require one interpretation or the other
+> in the spec.
```

Comment:
And I would highlight this...

```suggestion
> [!NOTE]
> During the Technical Preview, feedback on how the registry
> describes how _functions_ inherit resolved values and _options_
> and what requirements this specification should impose
> are highly desired.
```
```

Reply:
I'm working on a revised version of this, but note that the word "inherit" should be completely off-limits since it connotes object-oriented inheritance, which would just confuse the issue.